src/vm/contexts.rs

```diff
@@ -21,26 +28,38 @@ impl <'a> Environment <'a> {
     }
 }
 
+// Aaron: note -- only the global context will ever have DefinedFunctions
+//        so it is probably worthwhile to separate into 2 types.
+
```

Comment:
Probably a good idea -- goes along with the general Rust-ism of avoiding non-sensical instantiations by making them unrepresentable in the first place.

Reply:
Yep